src/utils/c_utils/src/string_c_utils.c

```diff
@@ -417,26 +420,35 @@ bool MallocAndSubstr(char** target, char* source, size_t len)
  *
  * @param dest The destination buffer.
  * @param src The source string to be copied.
- * @param destByteLen The size of the destination buffer in bytes.
+ * @param destByteLen The size of the destination buffer in bytes, including plus 1 for the null terminator.
  * @param numSrcCharsToCopy The number of source chars to copy. It will be truncated and null-terminated if >= destByteLen.
- * @return size_t The number of src chars copied. If < numSrcCharsToCopy, then it was truncated.
+ * @return int zero on success, or a negative number if there was an error.
```

Comment:
why'd you choose zero for success? I can easily see this occurring?

```c
if (!ADUC_Safe_StrCopyN(dest, src, len, tocopy)) {
  return false;
}

Reply:
Changed to return bool true as I think your suggestion is more natural here.

However, due to C's implicit type coercion, returning int 0 is just as fraught with danger as return bool true, depending on if whoever is using it assumes one or the other (and ignores the code and/or doxygen comments surfacing in the IDE or editor).

There's no gcc compiler flag I know of that will warn/error if coercion is going on, but clang-tidy should catch it via https://releases.llvm.org/13.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/readability-implicit-bool-conversion.html